.github/workflows/quemb_unittest.yml

```diff
@@ -18,20 +18,39 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+
     - name: Set up Python 3.9
       uses: actions/setup-python@v5
       with:
         python-version: "3.9"
+
     - uses: actions/cache@v4
       with:
         path: ${{ env.pythonLocation }}
         key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('dev-requirements.txt') }}
+
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install --upgrade --upgrade-strategy eager pytest
         if [ -f requirements.txt ]; then pip install --upgrade --upgrade-strategy eager -r requirements.txt; fi
         echo ${{ github.workspace }} > $(python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")/quemb.pth
+
     - name: Test with pytest
       run: |
-        pytest
+        cd tests
+        pytest --doctest-modules --junitxml=junit/quemb-test-results_${{ matrix.python-version }}.xml
+      if: always()
+
+    - name: Upload pytest junit results
+      uses: actions/upload-artifact@v4
+      with:
+        name: quemb-test-results_${{ matrix.python-version }}
+        path: tests/junit/quemb-test-results_${{ matrix.python-version }}.xml
+
+    - name: Upload pytest html results
+      uses: actions/upload-artifact@v4
+      with:
+        name: quemb-tests-coverage_${{ matrix.python-version }}
+        path: tests/htmlcov
+      if: always()
```

Comment:
No artifacts are uploaded since code coverage is not activated. See annotations in a [recent run](https://github.com/oimeitei/quemb/actions/runs/11443644078).